mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -168,9 +169,19 @@ class CrossValidator @Since("1.2.0") (@Since("1.4.0") override val uid: String)
       val validationDataset = sparkSession.createDataFrame(validation, schemaWithoutFold).cache()
       instr.logDebug(s"Train split $splitIndex with multiple sets of parameters.")
 
+      val sparkContext = sparkSession.sparkContext
+      val oldJobGroup = sparkContext.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID)
+      val cvJobGroup = s"${this.uid}_job_group"
+      sparkContext.setLocalProperty(SparkContext.SPARK_JOB_GROUP_ID, cvJobGroup)
+      @volatile var subTaskFailed = false
       // Fit models in a Future for training in parallel
       val foldMetricFutures = epm.zipWithIndex.map { case (paramMap, paramIndex) =>
         Future[Double] {
+          if (subTaskFailed) {
+            throw new RuntimeException(
```

Comment:
The exception doesn't matter here? so RuntimeException is OK? otherwise that's probably overly broad. At least, IllegalStateException or something. But if it's just going to kill the Future, OK

Reply:
Yes just for kill the future thread.
I already add message in the runtimeException. So I think it is fine.
`IllegalStateException` means "Signals that a method has been invoked at an illegal or inappropriate time" so it may be not a proper exception class name.